cpp/include/cudf/text/tokenize.hpp

```diff
@@ -0,0 +1,151 @@
+ * @param delimiter UTF-8 characters used to separate each string into tokens.
+ *                  The default of empty string will separate tokens using whitespace.
+ * @param mr Resource for allocating device memory.
+ * @return New strings columns of tokens.
+ */
+std::unique_ptr<column> tokenize( strings_column_view const& strings,
+                                  string_scalar const& delimiter = string_scalar{""},
+                                  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+/**
+ * @brief Returns a single column of strings by tokenizing the input strings
+ * column using multiple strings as delimiters.
+ *
+ * Tokens are found by locating delimiter(s) starting at the beginning of each string.
+ * Any consecutive delimiters found in a string are ignored.
+ * This means only non-empty tokens are returned.
+ *
+ * As each string is tokenized, the tokens are appended using input column row order
+ * to build the output column. That is, tokens found in input row[i] will be placed in
+ * the output column directly before tokens found in input row[i+1].
+ *
+ * Example:
+ * ```
+ * s = ["a", "b c", "d,e:f;"]
+ * d = [",",":",";"]
+ * t = tokenize(s,d)
+ * t is now ["a","b c","d","e","f"]
+ * ```
+ *
+ * All null row entries are ignored and the output contains all valid rows.
+ *
+ * @throw cudf::logic_error if the delimiters column is empty or contains nulls.
+ *
+ * @param strings Strings column to tokenize.
+ * @param delimiters Strings used to separate individual strings into tokens.
+ * @param mr Resource for allocating device memory.
+ * @return New strings columns of tokens.
+ */
+std::unique_ptr<column> tokenize( strings_column_view const& strings,
+                                  strings_column_view const& delimiters,
+                                  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+/**
+ * @brief Returns the number of tokens in each string of a strings column.
+ *
+ * The `delimiter` may be zero or more characters. If the `delimiter` is empty,
+ * whitespace (character code-point <= ' ') is used for identifying tokens.
+ * Also, any consecutive delimiters found in a string are ignored.
+ * This means that only empty strings or null rows will result in a token count of 0.
+ *
+ * Example:
+ * ```
+ * s = ["a", "b c", " ", "d e f"]
+ * t = token_count(s)
+ * t is now [1,2,0,3]
+ * ```
+ *
+ * All null row entries are ignored and the output contains all valid rows.
+ * The number of tokens for a null element is set to 0 in the output column.
+ *
+ * @param strings Strings column to use for this operation.
+ * @param delimiter Strings used to separate each string into tokens.
+ *                  The default of empty string will separate tokens using whitespace.
+ * @param mr Resource for allocating device memory.
+ * @return New INT32 column of token counts.
+ */
+std::unique_ptr<column> token_count( strings_column_view const& strings,
```

Comment:
Should this be `count_tokens` to make the function name a verb rather than a noun? I think most libcudf algorithm names are verbs.